molo/profiles/forms.py

```diff
@@ -162,10 +162,11 @@ def is_valid(self):
                 settings = SettingsProxy(site)
                 profile_settings = settings['profiles']['UserProfilesSettings']
                 number = self.data['mobile_number']
-                if number.startswith('0'):
-                    number = number[1:]
-                number = profile_settings.country_code + \
-                    number
+                if number != '':
```

Comment:
@nathanbegbie a better check will be `if number` which will ensure that you catch the `None` case too